R/utilities.R

```diff
@@ -4117,3 +4130,700 @@ supplement_maf <- function(incoming_maf,
   return(full_maf)
 }
+
+
+#' Classify FL samples into cFL/dFL subgroups.
+#'
+#' Use the random forest prediction model to assemble the binary matrix and use it to classify FL tummors into cFL/dFL
+#'
+#' @param these_samples_metadata The metadata data frame that contains sample_id column with ids for the samples to be classified.
+#' @param maf_data The MAF data frame to be used for matrix assembling. At least must contain the first 45 columns of standard MAF format.
+#' @param model The RF model. Classifier from the paper describing cFL is used. It is not recommended to change the value of this parameter.
+#' @param seq_type The seq_type of the samples. Only really used to retrerive mutations when maf data is not provided and to be retreived through GAMBLR. Defaults to genome.
+#' @param output The output to be returned after prediction is done. Can be one of predictoins, matrix, or both. Defaults to predictions.
+#'
+#' @return data frame with classification, binary matrix used in classification, or both
+#' @export
+#' @import tidyverse randomForest
+#'
+#' @examples
+#' test_meta <- get_gambl_metadata(case_set="tFL-study")
+#' predictions = classify_fl(these_samples_metadata = test_meta)
+#' predictions = classify_fl(these_samples_metadata = test_meta, output = "both")
+#'
+classify_fl <- function(
+    these_samples_metadata,
+    maf_data,
+    model = RFmodel_FL,
+    seq_type = "genome",
+    output = "predictions"
+) {
+
+    # Establish minimum required set of genes
+    req_features <- rownames(
+        model$importance
+    )
+
+    ssm_features <- req_features[!grepl("HOTSPOT|_TSS|inKATdomain|_intronic|_intron_1", req_features)]
+    ssm_features = gsub(
+        "_",
+        "-",
+        ssm_features
+    )
+    hotspot_features <- req_features[grepl("HOTSPOT|inKATdomain", req_features)]
+    ashm_features <- req_features[grepl("_TSS|_intronic|_intron_1", req_features)]
+    ashm_features_bed <- grch37_ashm_regions %>%
+        dplyr::mutate(
+            name = gsub("-", "_", grch37_ashm_regions$name)
+        ) %>%
+        dplyr::filter(name %in% ashm_features | name %in% c("PAX5_TSS_1", "SGK1_TSS_1"))
+
+    req_features <- gsub(
+        "HOTSPOT|_TSS|inKATdomain|_intronic|_intron_1",
+        "",
+        req_features
+    )
+    req_features <- gsub(
+        "_",
+        "-",
+        req_features
+    )
+    req_features <- sort(
+        unique(
+            req_features
+        )
+    )
+
+    if(missing(these_samples_metadata) & missing(maf_data)){
+        stop("Exiting. Please provide the sample metadata or maf data to use in classification.")
+    }else if (missing(maf_data)){
+       message(
+            "No maf data was provided. Retreiving SSMs using GAMBLR..."
+       )
+       maf_data =  get_ssm_by_samples(
+            these_samples_metadata = these_samples_metadata,
+            seq_type = seq_type,
+            subset_from_merge = TRUE,
+            augmented = FALSE
```

Comment:
Right, I did not use the augmented here because wanted the time points to be "independent", so to use only variants seen at the time of biopsy. In any case, if the user will prefer to use the augmented data, there is always an option to supply it directly with `maf_data` argument